src/Column/ColumnDefinitionParser.php

```diff
@@ -24,32 +24,50 @@ final class ColumnDefinitionParser extends \Yiisoft\Db\Syntax\ColumnDefinitionPa
         . '|\w*'
         . ')(?:\(([^)]+)\))?)(\[[\d\[\]]*\])?\s*/i';
 
-    public function parse(string $definition): array
+    protected function parseDefinition(string $definition): array
     {
         preg_match(self::TYPE_PATTERN, $definition, $matches);
 
         /** @var string $type */
         $type = $matches[3] ?? preg_replace('/\s*\(\d+\)/', '', $matches[1]);
-        $type = strtolower($type);
-        $info = ['type' => $type];
-
-        $typeDetails = $matches[4] ?? $matches[2] ?? '';
-
-        if ($typeDetails !== '') {
-            if ($type === 'enum') {
```

Comment:
'enum' here and in other drivers should be an abstract type and must be parsable.